src/lib/routes/load-user-request-route.ts

```diff
@@ -0,0 +1,212 @@
+import { FastifyInstance } from "fastify";
+import {
+	ProcessedDocument,
+	ProcessedDocumentChunk,
+	ProcessedDocumentSummary,
+	RegisteredDocument,
+	ResponseDocumentMatchReference,
+	UserRequest,
+} from "../common.js";
+import { registerCors } from "../handle-cors.js";
+import supabase from "../supabase.js";
+import { getUserRequestSchema } from "../json-schemas.js";
+
+function findSimilarityForChunk(
+	chunkId: number,
+	matches: Array<ResponseDocumentMatchReference>,
+): number {
+	const match = matches.find((match) => {
+		return match.processed_document_chunk_matches.some((chunkMatch) => {
+			return chunkMatch.processed_document_chunk_id === chunkId;
+		});
+	});
+
+	if (!match) {
+		return -1;
+	}
+
+	const chunkMatch = match.processed_document_chunk_matches.find(
+		(chunkMatch) => {
+			return chunkMatch.processed_document_chunk_id === chunkId;
+		},
+	);
+
+	return chunkMatch?.similarity ?? -1;
+}
+
+function findSimilarityForSummary(
+	summaryId: number,
+	matches: Array<ResponseDocumentMatchReference>,
+): number {
+	const match = matches.find((match) => {
+		return (
+			match.processed_document_summary_match.processed_document_summary_id ===
+			summaryId
+		);
+	});
+
+	if (!match) {
+		return -1;
+	}
+
+	return match.processed_document_summary_match.similarity;
+}
+
+export async function registerLoadUserRequestRoute(fastify: FastifyInstance) {
+	await fastify.register(
+		async (app, options, next) => {
+			await registerCors(app);
+
+			app.get(
+				"/:requestId",
+				{
+					schema: {
+						response: getUserRequestSchema,
+					},
+				},
+				async (request, reply) => {
+					const { requestId } = request.params as { requestId: string };
+
+					const { data, error } = await supabase
```

Comment:
Why not handle the error and just check for missing data?

Reply:
Error handling can definitely be improved here, yes. I just missed handling the error here, iterating on error handling is already a todo ticket.

Reply:
Working on that ticket